tests/test_api_requests.py

```diff
@@ -137,6 +139,124 @@ async def test_http_response_headers_request(
             {"browserHtml": True, "geolocation": "US", "javascript": True},
             True,
         ),
+        # ZYTE_API_ON_ALL_REQUESTS=False
```

Comment:
Since this is the default value, it's quite redundant with the test cases above that omits it.

My concern here is that the test cases is a bit harder to maintain because of it since changing/adding something would mean that 2 test cases are involved.

Reply:
Switched to a generator to keep the test cases without code repetition.